.github/workflows/build-and-push-test-image.yml

```diff
@@ -0,0 +1,96 @@
+name: Build and Push Test Image
+
+# This workflow builds and pushes the backend/test/Dockerfile.test image to quay.io
+# The image is tagged with the branch name to allow for branch-specific testing
+
+on:
+  push:
+    branches:
+      - master
+      - 'v*'
+      - stable
+  pull_request_target:
```

Comment:
I'd be careful with this - this workflow looks fine to me but just double/triple check this and ensure it can never execute user code (including reading the pr head)  lol